### PR TITLE
allow multiple operations per file

### PR DESCRIPTION
**Pull request by vasilich6107**

### CODE REVIEW

File: lib/generator.dart [hunk 1]
```diff
@@ -115,97 +115,108 @@ QueryDefinition generateQuery(
...
-      break;
-    case OperationType.mutation:
-      suffix = 'Mutation';
-      break;
-    case OperationType.query:
-    default:
-      suffix = 'Query';
-      break;
-  }
+  return operations.map((operation) {
```
Comment (comigor) [inline lib/generator.dart:121]: Can we extract this to another function? Maybe a `generateQueries` and another `generateQuery`.

File: lib/generator.dart [hunk 2]
```diff
@@ -115,97 +115,108 @@ QueryDefinition generateQuery(
...
+      context: context.sameTypeWithNoPath(),
+    );
+
+    DocumentNode(
+      definitions: document.definitions
+          // filtering unused operations
+          .where((e) => e is! OperationDefinitionNode || e == operation)
+          .toList(),
+    ).accept(visitor);
+    schema.accept(canonicalVisitor);
```
Comment (comigor) [inline lib/generator.dart:202]: Canonical visitor could ran only once, right?

Comment (vasilich6107) [reply lib/generator.dart:202]: Good catch. I'll look

**Comment by vasilich6107** [comment]
It seems that decoupling require a duplication as far as this two function are heavily coupled)
My PR is some kind of a quick fix)

Lately the generator will require a little bit refactoring)

**Pull request merged**